core/filter_api.php

```diff
@@ -1512,12 +1509,9 @@ function filter_get_bug_rows_query_clauses( array $p_filter, $p_project_id = nul
 	}
 
 	# creation date filter
-	if( ( 'on' == $t_filter[FILTER_PROPERTY_FILTER_BY_DATE_SUBMITTED] )
-			&& is_string( $t_filter[FILTER_PROPERTY_START_DATE_SUBMITTED] )
-			&& is_string( $t_filter[FILTER_PROPERTY_END_DATE_SUBMITTED] )
-			) {
-		$t_start_string = $t_filter[FILTER_PROPERTY_START_DATE_SUBMITTED] . ' 00:00:00';
-		$t_end_string = $t_filter[FILTER_PROPERTY_END_DATE_SUBMITTED] . ' 23:59:59';
+	if( ( 'on' == $t_filter[FILTER_PROPERTY_FILTER_BY_DATE_SUBMITTED] )	) {
+		$t_start_string = filter_read_last_updated_start_date( $t_filter );
```

Comment:
This function refers to last updated even though the if statement refers to date submitted.